LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/BasicsStation/JsonHandlers/LnsStationConfiguration.cs

```diff
@@ -174,19 +179,22 @@ select JoinEui.Parse(eui))
                               JsonReader.Property("freq_range", from r in JsonReader.Array(JsonReader.UInt32())
                                                                 select (new Hertz(r[0]), new Hertz(r[1]))),
                               JsonReader.Property("DRs",
-                                  JsonReader.Array(from e in JsonReader.Tuple(JsonReader.Either(from n in JsonReader.Int32().Validate(n => n == 0)
-                                                                                                select FskSpreadingFactor,
-                                                                                                JsonReader.Int32().AsEnum(n => unchecked((SpreadingFactor)n))),
+                                  JsonReader.Array(from e in JsonReader.Tuple(JsonReader.Either(from n in JsonReader.Int32().Validate(n => n is 0 or -1)
+                                                                                                select (SpreadingFactor)n,
+                                                                                                JsonReader.UInt32().AsEnum(n => unchecked((SpreadingFactor)n))),
```

Comment:
This can be undone, no?

```suggestion
                                                                                                JsonReader.Int32().AsEnum(n => unchecked((SpreadingFactor)n))),
```